share/wake/lib/core/result.wake

```diff
@@ -95,6 +95,14 @@ export def rmap (fn: a => b): Result a fail => Result b fail = match _
     Pass a -> Pass (fn a)
     Fail f -> Fail f
 
+# rmapError: apply a function to a Failing-ing result
+#
+#   rmapError (_+1) (Pass 123) = Pass 123
+#   rmapError (_+1) (Fail 123) = Fail 124
+export def rmapError (fn: a => b): Result c a => Result c b = match _
+    Pass a -> Pass a
+    Fail f -> Fail (fn f)
```

Comment:
This is definitely a more comfortable signature than `rmapFail`, but not everything would be an `Error` given the freer typing, so the name's a bit misleading.  It might also be nice to define it in terms of `rmapFail (fn _ | Fail)`, or if you keep the raw `match` for simplicity please at least make the `Pass a` binding use the same letter as the `Result c x` annotation.